src/main/java/com/wepay/zktools/clustermgr/internal/ClusterManagerImpl.java

```diff
@@ -253,7 +253,7 @@ public void manage(ManagedServer server) throws ClusterManagerException {
 
                             // retry if the partitionAssignment failed to update. (May happen because of
                             // race-condition.)
-                            if (updatePartitionAssignment(version, assignment.cversion, membership, assignment)) {
+                            if (updatePartitionAssignment(version, assignment.cversion, membership, assignment, true)) {
```

Comment:
added a `forceUpdate` flag to updatePartitionAssignment(....) to handle scenario like if preferred partitions are changed for a server id (that is not the min Server Id) in the cluster, then that change will get reflected.

Imagine a scenario that has Server1 (ClusterManager1) and Server2 (ClusterManager2), if we try to re-manage Server 2 (by add/remove preferred partition), then this will trigger:
 - UpdateServer (by ClusterManager 1) with old Version and old cVersion ====> No action will be taken.
 - UpdateServer (by clusterManager 2) with old Version and old cVersion ====> No action will be taken.
 - UpdatePreferredPartitions (by ClusterManager 2) with old Version and old cVersion ====> if we don't consider the flag, `isCoordinate` will return false and this change (add/remove preferred partition of Server 2) will not be reflected in the cluster.